cni/pkg/nodeagent/informers.go

```diff
@@ -262,20 +262,6 @@ func (s *InformerHandlers) reconcilePod(input any) error {
 			log.Debugf("RemovePodFromMesh returned: %v", err)
 			// we ignore errors here as we don't want this event to be retried by the queue.
 		} else {
-			// If oldpod != ready && newpod != ready, but the ambient annotation was added,
-			// then assume this event was generated by the CNI plugin labeling the pod on startup,
-			// and skip the event.
-			//
-			// This isn't perfect (someone could manually annotate an unready pod,
-			// then install Istio, then the pod goes ready, and we'd miss capture) - but that
-			// seems vanishingly unlikely
```

Comment:
Yeah - This was originally designed to swallow the case where the plugin handler annotating the pod would generate a (spurious) informer event we should ignore.

but given `changeNeeded` check I think it's redundant now.